### PR TITLE
Sphinx doc enhancement

### PR DESCRIPTION
**Pull request by TickyWill**
Based on sphinx_technical_documentation branch, this is a tentative enhancement of  the documentation building .
The present branch include README update and all docstrings in BiblioMeter modifyed to comply with sphinx and markdown rules.
See the README file to build the documentation working in this branch (sphinx_doc_enhancement) and to view the result and give me a feedback.
Thanks in advance.

### CODE REVIEW

**Comment by BaptisteRefalo** [comment]
Hello, since this branch is based on "sphinx_technical_documentation" with its PR pointing towards "v5.0.0" directly, the "sphinx_technical_documentation" branch is no longer needed, is it not ?
Shouldn't it be more interesting to point this PR towards "sphinx_technical_documentation" since it is enhancements on this branch and merge the "sphinx_technical_documentation" branch in "v5.0.0" afterwards (with the other existing PR)? It would certainly ease the reviews :)

**Comment by BaptisteRefalo** [comment]
The "docbuild" folder should be ignored in the ".ignore" file (the html files will not be put here)

File: docs/make.bat [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+pushd %~dp0
+
+REM Command file for Sphinx documentation
+
+if "%SPHINXBUILD%" == "" (
+	set SPHINXBUILD=sphinx-build
+)
+set SOURCEDIR=source
+set BUILDDIR=docbuild
```
Comment (BaptisteRefalo) [inline docs/make.bat:11]:
```suggestion
set BUILDDIR=build
```
Either way, this folder must be ignored :)

File: docs/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line, and also
+# from the environment for the first two.
+SPHINXOPTS    ?=
+SPHINXBUILD   ?= sphinx-build
+SOURCEDIR     = source
+BUILDDIR      = docbuild
```
Comment (BaptisteRefalo) [inline docs/Makefile:9]: "build" or "docbuild": generally, "build" is the name used

File: docs/source/conf.py
[the remark this answers is not in the document]
Comment (TickyWill) [reply docs/source/conf.py:1]: OK. I'll have tried but not yet succeeded. I will try again.
Many thanks.

File: docs/Makefile (hunk 2 above)
Comment (TickyWill) [reply docs/Makefile:9]: Yes, you are right. it was a resudual change of trying to make the build documentation available on BiblioMeter repository. I'll come back to "build".
Many thanks

**Pull request merged**